File: programming files/sum_even_fibonacci.java
// Bugs introduced: if statement & int temp declaration
// By IC

public class sum_even_fibonacci {
    public static void main(String[] args) {
        int a = 0, b = 1;
        int sumEven = 0;
        while (b < 4000000) {
            if if (b % 2 == 0) {  // should be: if (b % 2 == 0) {
                sumEven += b;
            }
            int temp = b;  // should be int temp = b;
            b = a + b;
            a = temp;
        }
        System.out.println(sumEven);
    }
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.py
# BUG INTRODUCED
a, b = 0, 1
sum_even = 0
while b < 4000000:
    # check if even
    if b % 3 == 0:
        sum_even += b
    # a becomes b, b becomes sum of a and b
    a, b = b, a+b

print(sum_even)

# answer should be 4613732
